### PR TITLE
Ports: Update freeciv to version 3.1.1

### PR DESCRIPTION
**Pull request by kennethmyhra**
- Updates `freeciv` to version 3.1.1.
- Sets '--enable-fcdb=no'. The default has been changed from 'no' in previous versions to 'sqlite3' in this version.
- Formats the package script according to our ruleset.

### CODE REVIEW

File: Ports/freeciv/package.sh [hunk 1]
```diff
@@ -30,3 +30,11 @@ launcher_name='Freeciv'
...
 launcher_command='/usr/local/bin/freeciv-sdl2'
 icon_file='windows/client.ico'
+
+post_install() {
+    echo
+    echo "Freeciv version ${version} is installed!"
+    echo
+    echo "Release notes: https://freeciv.fandom.com/wiki/NEWS-${version}"
+    echo
+}
```
Comment (timschumi) [inline Ports/freeciv/package.sh:40]: Why though? Nobody else does this. :yakthonk:

Comment (kennethmyhra) [reply Ports/freeciv/package.sh:40]: We do this for a few ports. I can drop it though if we do not want it.

Comment (kennethmyhra) [reply Ports/freeciv/package.sh:40]: Or maybe it is just the DOSbox staging that has this exact output. (I added that as well). Let me know if you want me to drop it and I'll drop it.

Comment (timschumi) [reply Ports/freeciv/package.sh:40]: I'd rather remove it, "port X version Y is now installed" is already mentioned by the ports framework itself, and the release notes are just moderately helpful for someone who just installed the version in question (compared to someone who wants to know whether it's worth to make the effort and upgrade).

Comment (kennethmyhra) [reply Ports/freeciv/package.sh]: Done

**Review by timschumi — CHANGES_REQUESTED** [review]
Currently does not build, at least on my machine:

```
[freeciv/build] ld: /serenity/Toolchain/../Build/x86_64/Root/usr/lib/crt0.o: undefined reference to symbol '__begin_atexit_locking'
[freeciv/build] ld: /serenity/Toolchain/../Build/x86_64/Root/usr/lib/libc.so: error adding symbols: DSO missing from command line
```

**Comment by kennethmyhra** [comment]
Figured out why this was failing: The default value for the option `--enable-fcdb` has changed from 'no' in previous versions to 'sqlite3' in this version.

Still not sure why it worked on some machines, like my laptop, but I guess that can be a future task for figuring out how to enable sqlite3 for this port.

**Review by timschumi — APPROVED** [review]

**Pull request merged**